pkg/visitor/subtree.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2023 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package visitor
+
+import (
+	"context"
+
+	"github.com/apigee/registry/pkg/names"
+	"github.com/apigee/registry/rpc"
+)
+
+// SubtreeHandlers is a Visitor implementation that can be passed to the
+// Visit() function. It will run the handlers in Visitor against each
+// resource directly matched in Visit() as well as each resource that is
+// a descendant of those resources. (For example, if the pattern matches
+// a Version, the SubtreeVisitor will be run against all of that Versions's
+// Artifacts and Specs, and the Spec's Artifacts as well.)
+// Only the current revisions of Specs and Deployments are visited.
+type SubtreeHandlers struct {
```

Comment:
Could we call this "SubtreeVisitor" instead of "SubtreeHandlers"? I see both names used in the comment and I think "SubtreeVisitor" more clearly reminds me that this is a visitor implementation that does something special (visits subtrees). Thinking out loud, I'd say this is a kind of meta-visitor, a visitor based on a visitor, that applies its underlying visitor in a specified way.

Reply:
Sure. It's both in the comments because I renamed it from SubtreeVisitor to SubtreeHandlers and missed a couple places. FWIW, I renamed it because "visitor" gets overloaded because of the package name stutter and we call the functions themselves handlers, but I'll change it as you suggest.

Reply:
Thanks! I think it's helpful to describe this as a visitor because it implements the Visitor interface. The package name stutter doesn't bother me much, [context.Context](https://pkg.go.dev/context), of all things, is an easy-to-find example of that!